integration/spark-datasource/src/main/scala/org/apache/spark/sql/carbondata/execution/datasources/CarbonSparkDataSourceUtil.scala

```diff
@@ -198,7 +249,20 @@ object CarbonSparkDataSourceUtil {
       val dataType = convertSparkToCarbonDataType(field.dataType)
       dataType match {
         case s: CarbonStructType =>
-          new Field(field.name, s, s.getFields)
+          val subFields = prepareSubFields(field.name, s)
```

Comment:
Why do we need to create subfields here?

Reply:
moved the preparation to Field class